src/core/thread/mle.cpp

```diff
@@ -4418,6 +4428,66 @@ void Mle::DelayedResponseMetadata::RemoveFrom(Message &aMessage) const
     OT_UNUSED_VARIABLE(error);
 }
 
+#if OPENTHREAD_CONFIG_TMF_DUA_REGISTRATION_ENABLE
+void Mle::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefixState aState)
```

Comment:
Why is this logic in MLE? Can we put Domain Unicast Address management in a separate source file? Maybe in `backbone_router` given that Domains are related to BBRs?

Reply:
Moved to the separate dua_manager module as discussed. Please review.